pootle/apps/pootle_store/migrations/0027_unit_created_by_squashed_0055_fill_unit_source_data.py

```diff
@@ -0,0 +1,432 @@
+    pending = states.get(name="pending")
+    accepted = states.get(name="accepted")
+    rejected = states.get(name="rejected")
+    suggestions.filter(tmp_state=OLDSuggestionStates.PENDING).update(state_id=pending.id)
+    suggestions.filter(tmp_state=OLDSuggestionStates.ACCEPTED).update(state_id=accepted.id)
+    suggestions.filter(tmp_state=OLDSuggestionStates.REJECTED).update(state_id=rejected.id)
+
+
+def clean_abs_file_paths(apps, schema_editor):
+    """Replace wrong absolute store file paths by proper relative paths
+
+    built based on store.pootle_path values.
+    """
+    store_model = apps.get_model("pootle_store.Store")
+    stores = store_model.objects.filter(
+        translation_project__project__treestyle="nongnu")
+    stores = stores.filter(file__startswith="/").only("file", "pootle_path")
+    to_update = []
+    for store in stores.iterator():
+        lang, prj, d, fn = split_pootle_path(store.pootle_path)
+        store.file = os.path.join(prj, lang, d, fn)
+        to_update.append(store)
+    if to_update:
+        result = Batch(store_model, batch_size=500).update(
+            to_update,
+            update_fields=["file"],
+            reduces=False)
+        logger.debug("Cleaned %s store paths" % result)
+
+
+def remove_fields_with_sql(apps, schema_editor):
+    Unit = apps.get_model("pootle_store.Unit")
+    cursor = schema_editor.connection.cursor()
+    for col in REMOVED_UNIT_COLS:
+        field = Unit._meta.get_field(col)
+        if field.remote_field:
+            fk_names = schema_editor._constraint_names(
+                Unit, [field.column], foreign_key=True)
+            for fk_name in fk_names:
+                cursor.execute(
+                    schema_editor._delete_constraint_sql(
+                        schema_editor.sql_delete_fk, Unit, fk_name))
+    cursor.execute(
+        UNIT_DELETE_COLS_SQL
+        % (", ".join(
+            [("DROP COLUMN `%s`" % col)
+             for col in REMOVED_UNIT_COLS])))
+
+
+def remove_fields(apps, schema_editor):
+    if schema_editor.connection.vendor == "mysql" and settings.POOTLE_SQL_MIGRATIONS:
+        remove_fields_with_sql(apps, schema_editor)
+
+
+class RemoveFieldIfExists(migrations.RemoveField):
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        if schema_editor.connection.vendor == "mysql" and settings.POOTLE_SQL_MIGRATIONS:
+            return
+        super(RemoveFieldIfExists, self).database_forwards(
+            app_label, schema_editor, from_state, to_state)
+
+
+class Migration(migrations.Migration):
+
+    replaces = [
+        (b'pootle_store', '0027_unit_created_by'),
+        (b'pootle_store', '0028_set_created_by'),
+        (b'pootle_store', '0029_unit_tablename'),
+        (b'pootle_store', '0030_store_tablename'),
+        (b'pootle_store', '0031_suggestion_tablename'),
+        (b'pootle_store', '0032_qc_tablename'),
+        (b'pootle_store', '0033_conditionally_remove_unit_created_by'),
```

Comment:
Once `created_by` is moved to `UnitSource` is it not ok to remove this field from `Unit`?